### PR TITLE
fixed another instance of the tablet getting stuck on the help screen

### PR DESCRIPTION
**Pull request by druiz17**
fixed another instance of the tablet getting stuck on the help screen. I

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan
1
- Do a clean install and do not have your sandbox running
- Start interface with in HMD mode
- Open the tablet and you should be on the help screen
- Pressing the x button on the tablet should bring you back to the home screen, pressing it again should close the tablet.
- Open tablet again and you should be on the home-screen

2
- Do a clean install and do not have your sandbox running
- Start interface with in HMD mode
- Quickly switch to desktop mode
- Once in desktop mode, the button in the toolbar that should be highlighted is the bubble button.
- Without clicking anything switch back to HMD mode
- Open the tablet and you should be on the home screen.
- Pressing the x button on the tablet should bring you back to the home screen, pressing it again should close the tablet.
- Open tablet again and you should be on the home-screen

3
- Do a clean install and do not have your sandbox running
- Start interface with in HMD mode
- Quickly switch to desktop mode
- Once in desktop mode, the button in the toolbar that should be highlighted is the bubble button.
- Click on help button to bring up the help page
- With the help page open, switch to HMD mode
- Open the tablet and you should be on the home screen.
- Pressing the x button on the tablet should bring you back to the home screen, pressing it again should close the tablet.
- Open tablet again and you should be on the home-screen

**Comment by YuppyPlays** [comment]
Will test once built :)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1754/HighFidelity-Beta-PR10990-06a69c8bd72ca1f01cc3e60303530249ae590e34.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1754/HighFidelity-Beta-PR10990-06a69c8bd72ca1f01cc3e60303530249ae590e34.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

1
Do a clean install and do not have your sandbox running
Start interface with in HMD mode
Open the tablet and you should be on the help screen
Pressing the x button on the tablet should bring you back to the home screen, pressing it again should close the tablet.
Open tablet again and you should be on the home-screen
**Pass**



2
Do a clean install and do not have your sandbox running
Start interface with in HMD mode
Quickly switch to desktop mode
Once in desktop mode, the button in the toolbar that should be highlighted is the bubble button.
Without clicking anything switch back to HMD mode
Open the tablet and you should be on the home screen.
Pressing the x button on the tablet should bring you back to the home screen, pressing it again should close the tablet.
Open tablet again and you should be on the home-screen
**Pass**

3
Do a clean install and do not have your sandbox running
Start interface with in HMD mode
Quickly switch to desktop mode
Once in desktop mode, the button in the toolbar that should be highlighted is the bubble button.
Click on help button to bring up the help page
With the help page open, switch to HMD mode
Open the tablet and you should be on the home screen.
Pressing the x button on the tablet should bring you back to the home screen, pressing it again should close the tablet. **Fail? Switching from Desktop with Help page open to HMD works as test plan describes but pressing X on the home screen just closes the tablet, there is no second press of it unless I open it back up?... Not sure if wording issue or an actual fail**
Open tablet again and you should be on the home-screen
**Fail, see bold text above**

**Comment by druiz17** [comment]
@YuppyPlays sorry, when switching from desktop mode. Since you are already on the home screen, pressing x should close it. If you open the tablet again you should still be on the home screen.

**Comment by YuppyPlays** [comment]
@druiz17 
Ah, then I believe that makes this a **Pass** :) No worries! thank you for letting me know.

File: libraries/ui/src/ui/TabletScriptingInterface.cpp [hunk 1]
```diff
@@ -226,9 +226,12 @@ void TabletProxy::setToolbarMode(bool toolbarMode) {
...
         connect(tabletRootWindow, &QmlWindowClass::fromQml, this, &TabletProxy::fromQml);
+        loadHomeScreen(true);
+        emit screenChanged(QVariant("Home"), QVariant(TABLET_SOURCE_URL));
     } else {
         removeButtonsFromToolbar();
         addButtonsToHomeScreen();
+        loadHomeScreen(true);
         emit screenChanged(QVariant("Home"), QVariant(TABLET_SOURCE_URL));
 
         // destroy desktop window
```
Comment (AndrewMeadows) [inline libraries/ui/src/ui/TabletScriptingInterface.cpp]: These two lines are in both cases of the `if + else` logic.  Maybe pull them out and just call them after if-ing and else-ing?

```
         loadHomeScreen(true);
         emit screenChanged(QVariant("Home"), QVariant(TABLET_SOURCE_URL));
```

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1771/HighFidelity-Beta-PR10990-fdfeab77eb919568a5d3444c6aea6b76235dc8ec.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1771/HighFidelity-Beta-PR10990-fdfeab77eb919568a5d3444c6aea6b76235dc8ec.exe

**Pull request merged**